### PR TITLE
Implement motor currents reading

### PR DESCRIPTION
**Pull request by isorrentino**

### CODE REVIEW

File: src/RobotInterface/YarpImplementation/src/YarpSensorBridge.cpp [hunk 1]
```diff
@@ -397,3 +397,31 @@ bool YarpSensorBridge::getThreeAxisForceTorqueSensorsList(std::vector<std::strin
...
+}
+
+bool YarpSensorBridge::getMotorCurrents(Eigen::Ref<Eigen::VectorXd> jointCurrents,
+                                         double* receiveTimeInSeconds)
+{
+
+    jointCurrents = yarp::eigen::toEigen(m_pimpl->controlBoardRemapperMeasures.jointCurrents);
+    receiveTimeInSeconds = &m_pimpl->controlBoardRemapperMeasures.receivedTimeInSeconds;
+    return true;
+}
```
Comment (GiulioRomualdi) [inline src/RobotInterface/YarpImplementation/src/YarpSensorBridge.cpp:427]: End of line

File: src/RobotInterface/YarpImplementation/src/YarpSensorBridge.cpp [hunk 2]
```diff
@@ -397,3 +397,31 @@ bool YarpSensorBridge::getThreeAxisForceTorqueSensorsList(std::vector<std::strin
...
+bool YarpSensorBridge::getMotorCurrent(const std::string& jointName,
+                                        double& jointCurrent,
+                                        double* receiveTimeInSeconds)
+{
+    int idx;
+    if (!m_pimpl->getIndexFromVector(m_pimpl->metaData.sensorsList.jointsList,
+                                     jointName, idx))
+    {
+        std::cerr << "[YarpSensorBridge::getJointVelocity] " << jointName
+                  <<  " could not be found in the configured list of joints" << std::endl;
```
Comment (prashanthr05) [inline src/RobotInterface/YarpImplementation/src/YarpSensorBridge.cpp:410]:
```suggestion
        std::cerr << "[YarpSensorBridge::getMotorCurrent] " << jointName
                  <<  " could not be found in the configured list of joints" << std::endl;
```

**Comment by prashanthr05** [comment]
@isorrentino please be aware of this PR https://github.com/dic-iit/bipedal-locomotion-framework/pull/191

Whichever gets merged first, another might have to fix merge conflicts ;D

File: src/RobotInterface/YarpImplementation/include/BipedalLocomotion/RobotInterface/YarpSensorBridgeImpl.h [hunk 3]
```diff
@@ -1200,6 +1206,12 @@ struct YarpSensorBridge::Impl
                 std::cerr << logPrefix << " NAN values read from encoders interface, use previous measurement" << std::endl;
                 return false;
             }
+
+            if (std::binary_search(controlBoardRemapperMeasures.jointCurrents.begin(), controlBoardRemapperMeasures.jointCurrents.end(), NAN))
+            {
+                std::cerr << logPrefix << " NAN values read from current control interface, use previous measurement" << std::endl;
+                return false;
+            }
```
Comment (S-Dafarra) [inline src/RobotInterface/YarpImplementation/include/BipedalLocomotion/RobotInterface/YarpSensorBridgeImpl.h:1214]: I think that this should be changed according to #191. Indeed, by reading this code I opened #188 :grin:

File: src/RobotInterface/include/BipedalLocomotion/RobotInterface/ISensorBridge.h [hunk 4]
```diff
@@ -302,42 +302,90 @@ class ISensorBridge
...
      * @param[in] sensorLists SensorLists object holding list of connected sensor devices
      */
-    virtual bool populateSensorBridgeMetaDataFromConfig(std::weak_ptr<BipedalLocomotion::ParametersHandler::IParametersHandler> handler,
-                                                        SensorBridgeMetaData& sensorBridgeMetaData) { return true; };
+    virtual bool populateSensorBridgeMetaDataFromConfig(
+        std::weak_ptr<BipedalLocomotion::ParametersHandler::IParametersHandler> handler,
+        SensorBridgeMetaData& sensorBridgeMetaData)
+    {
+        return true;
+    };
```
Comment (S-Dafarra) [inline src/RobotInterface/include/BipedalLocomotion/RobotInterface/ISensorBridge.h:357]: Probably you can revert these changes, just to have the PR cleaner. In addition, the indentation is weird

Comment (isorrentino) [reply src/RobotInterface/include/BipedalLocomotion/RobotInterface/ISensorBridge.h]: Done!

**Review by S-Dafarra — APPROVED** [review]
